### ISSUE TITLE
manylinux1: disable dynamic engine support

### ISSUE THREAD

**Issue by tiran** [pyca/cryptography#3810]
It looks like my assumption in comment https://github.com/pyca/cryptography/issues/3804#issuecomment-317386649 was wrong. The uWSGI issue is not caused by a dynamic engine plugin. It may still be a good idea to compile OpenSSL with ``OPENSSL_NO_DYNAMIC_ENGINE`` to disable dynamic engine support. Engines such as ``engine_pkcs11`` from OpenSC are linked against system's libcrypto.so. I'm pretty sure such an engine is going to cause trouble when used with manylinux1 OpenSSL.

**Comment by tiran** [comment]
```
# ldd /usr/lib64/engines-1.1/libpkcs11.so
        linux-vdso.so.1 (0x00007fff066bc000)
        libcrypto.so.1.1 => /lib64/libcrypto.so.1.1 (0x00007f53ac12c000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f53abf28000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f53abb57000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f53ab940000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f53ab721000)
        /lib64/ld-linux-x86-64.so.2 (0x000056255e473000)
```

**Comment by reaperhulk** [comment]
Don't we use dynamic engines to load the osrandom engine?

**Comment by tiran** [comment]
Good point! I think osrandom engine qualifies as a static engine because it is statically linked into a shared library. I'm specifically talking about the DSO engine, which can load shared libraries with OpenSSL engines. I think the option disable-dynamic-engine only disables DSO.

**Comment by reaperhulk** [comment]
Worth an experiment then!

**Comment by alex** [comment]
Solved by #3810
